### ISSUE TITLE
[Feature Request] Add traffic optimization software to boost the speed

### ISSUE THREAD

**Issue by HowardMei** [jlund/streisand#723]
There are several traffic optimization suites good for speed:
Server & Client:  [kcptun](https://github.com/xtaci/kcptun)  
Server Side: [TCP BBR](https://blog.jie.today/2016/12/09/using-bbr-congestion-control-to-improve-connection-speed/) 

Could we have them as an option please?

**Comment by cpu** [comment]
Hi @HowardMei - this is a duplicate of https://github.com/jlund/streisand/issues/140, so I'm going to close it. Thanks for the suggestion, please follow #140 for that discussion.

**Comment by cpu** [comment]
Apologies! I commented/closed the wrong feature request. Reopening this one.

**Comment by cpu** [comment]
Looking at this again, the TCP BBR request is a duplicate of https://github.com/jlund/streisand/issues/546 - which (I believe) is blocked on a kernel dep.

**Comment by HowardMei** [comment]
Yeah. It demands kernel 4.9+, but could it be a default off option?
It's not easy do do testing or CI though.

**Comment by nickolasclarke** [comment]
@HowardMei is there an advantage of one over the other in terms of performance or featureset? Given that KCP is a standalone project vs BBR, which is an accepted commit to the a future kernel, it seems that from a maintenance standpoint that BBR would be the better option once it is part of a mainstream kernel. It seems that the 18.04 LTS will likely have a kernel version of 4.10+. Once most platforms make 18.04 available as a image option, Streisand could kick over to that and have this by default. 

IMO, that probably makes the most sense, given the time commitment that the core contributors can put towards this project at the moment. Others who want the enhanced performance from either option can modify their installations until BBR comes standard.

**Comment by HowardMei** [comment]
@nickolasclarke  BBR is supported in kernel 4.9+. I chose to implement BBR mode either. 
It's in: 
https://github.com/HowardMei/dostreisand/tree/develop/playbooks/roles/kernel-upgrade

I also changed some other parts like Vagrantfile/SSHD port that may not be appropriate for pull request. 
Therefore, I leave it to you guys to do the cherry-pick.

**Comment by cpu** [comment]
> I also changed some other parts like Vagrantfile/SSHD port that may not be appropriate for pull request. Therefore, I leave it to you guys to do the cherry-pick.

In general I have to take the difficult position of sharing that our team is very small - we likely don't have the time to pick through commits on a fork to try and cherry-pick a feature commit. If this is a feature you really want to see in the mainline Streisand you'll have to do more of the heavy-lifting to make it as easy as possible for us to review the work & merge it. Chances are you're the expert on your own work and would be able to pick apart the needed pieces much easier than us! Sending a pull request would be the first step! Thanks for understanding.

**Comment by HowardMei** [comment]
@cpu  Understandable.  I'll reconsider.

**Comment by cpu** [comment]
Closing in favour of https://github.com/jlund/streisand-discussions/issues/3.
